tests/test_bootstrap.py

```diff
@@ -100,33 +102,92 @@ def test_build_dist_dirs(self):
             bs.get_common_dir().endswith("pythonforandroid/bootstraps/common")
         )
 
-    def test_list_bootstraps(self):
+    def test_all_bootstraps(self):
         """A test which will initialize a bootstrap and will check if the
         method :meth:`~pythonforandroid.bootstrap.Bootstrap.list_bootstraps`
         returns the expected values, which should be: `empty", `service_only`,
         `webview` and `sdl2`
         """
         expected_bootstraps = {"empty", "service_only", "webview", "sdl2"}
-        set_of_bootstraps = set(Bootstrap().list_bootstraps())
+        set_of_bootstraps = Bootstrap().all_bootstraps()
         self.assertEqual(
             expected_bootstraps, expected_bootstraps & set_of_bootstraps
         )
         self.assertEqual(len(expected_bootstraps), len(set_of_bootstraps))
 
+    def test_expand_dependencies(self):
+        # Test depenency expansion of a recipe with no alternatives:
+        expanded_result_1 = expand_dependencies(["pysdl2"], self.ctx)
+        assert({"sdl2", "pysdl2", "python3"} in
+               [set(s) for s in expanded_result_1])
+
+        # Test all alternatives are listed (they won't have dependencies
+        # expanded since expand_dependencies() is too simplistic):
+        expanded_result_2 = expand_dependencies([("pysdl2", "kivy")], self.ctx)
+        assert([["pysdl2"], ["kivy"]] == expanded_result_2)
+
     def test_get_bootstraps_from_recipes(self):
-        """A test which will initialize a bootstrap and will check if the
-        method :meth:`~pythonforandroid.bootstrap.Bootstrap.
-        get_bootstraps_from_recipes` returns the expected values
-        """
-        recipes_sdl2 = {"sdl2", "python3", "kivy"}
-        bs = Bootstrap().get_bootstrap_from_recipes(recipes_sdl2, self.ctx)
+        import pythonforandroid.recipe
+        original_get_recipe = pythonforandroid.recipe.Recipe.get_recipe
+
+        @mock.patch("pythonforandroid.recipe.Recipe.get_recipe")
+        def do_test_get_bootstraps_from_recipes(mock_get_recipe):
+            """A test which will initialize a bootstrap and will check if the
+            method :meth:`~pythonforandroid.bootstrap.Bootstrap.
+            get_bootstraps_from_recipes` returns the expected values
+            """
+
+            # Use original get_recipe() until we need something else later:
+            mock_get_recipe.side_effect = original_get_recipe
```

Comment:
then if you don't need it mocked now, use the mock.patch context manager rather than the function decorator